README.md

```diff
@@ -242,6 +242,214 @@ remote_state = {
+the supported helper functions is `find_in_parent_folders`, which returns the path to the first `.terragrunt` file it
+finds in the parent folders above the current `.terragrunt` file. 
+
+Example:
+
+```hcl
+parent = {
+  path = "${find_in_parent_folders()}"
+}
+```
+
+If you ran this in `qa/my-app/.terragrunt`, this would automatically set `path` to `../../.terragrunt`. You will almost
+always want to use this function, as it allows you to copy/paste the same `.terragrunt` file to all child folders with
+no changes.
+
+### path_relative_to_parent helper
+
+Another helper function supported by Terragrunt is `path_relative_to_parent`, which returns the relative path between 
+the current `.terragrunt` file and the path specified in its `parent` block. For example, in the root `.terragrunt` 
+file, you could do the following:
+ 
+```hcl
+# Configure Terragrunt to use DynamoDB for locking
+lock = {
+  backend = "dynamodb"
+  config {
+    state_file_id = "${path_relative_to_parent()}"
+  }
+}
+
+# Configure Terragrunt to automatically store tfstate files in an S3 bucket
+remote_state = {
+  backend = "s3"
+  config {
+    encrypt = "true"
+    bucket = "my-bucket"
+    key = "${path_relative_to_parent()}/terraform.tfstate"
+    region = "us-east-1"
+  }
+}
+``` 
+
+Using the configuration above, each child `.terragrunt` file will get a unique path for its `state_file_id` and `key` 
+settings. For example, in `qa/my-app/.terragrunt`, the `state_file_id` will resolve to `qa/my-app` and the `key` will
+resolve to `qa/my-app/terraform.tfstate`.  
+
+You will almost always want to use this helper too. The only time you may want to specify the `state_file_id` or `key` 
+manually is if you moved a child folder. In that case, to ensure it can reuse its old state and lock, you may want to 
+hard-code the `state_file_id` and `key` to the old file path. However, a safer approach would be to move the state 
+files themselves to match the new location of the child folder, as that makes things more consistent!
+
+### Overriding settings from the parent
+
+Any settings in the child `.terragrunt` file will override the settings from the parent. For example, imagine if
+`qa/my-app/.terragrunt` had the following contents:
+ 
+```hcl
+parent = {
+  path = "${find_in_parent_folders()}"
+}
+
+remote_state = {
+  backend = "s3"
+  config {
+    encrypt = "true"
+    bucket = "some-other-bucket"
+    key = "/foo/bar/terraform.tfstate"
+    region = "us-west-2"
+  }
+}
+``` 
+
+The result is that when you run `terragrunt` commands in the `qa/my-app` folder, you get the `lock` settings from the 
+parent, but the `remote_state` settings of the child. 
+
+**Note**: only one level of inheritance is allowed. If `foo/.terragrunt` is specified as the `parent` of 
+`foo/bar/.terragrunt`, then `foo/.terragrunt` may NOT itself specify a `parent`. Otherwise, reasoning about helper
+functions such as `path_relative_to_parent` gets too complicated.
```

Comment:
This looks great overall, but I am left a little confused on the relationship between each of the interpolation functions and whether they are in a parent or child `.terragrunt` file. If I was reading your example correctly, the interpolation function itself, not its value, was inherited by child `.terragrunt` files, which is an awkward concept. Wouldn't it make more sense for the parent to simply omit the property you want each child to specify? In fact, you could even specify a value like `path = "${get_from_child()}"` to make explicit that this value should be specified by a child (though this probably isn't necessary).

Intuitively, I expect all interpolation functions to be resolved in the template they're in, and only the static value to be inherited. 

Also, given that we only support one level of inheritance (a good decision IMO), should we replace the concept of a "parent" folder with a "root" folder to make this more explicit?